packages/Operators/test/tstL2Projection.cpp

```diff
@@ -0,0 +1,186 @@
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*! 
+ * \file tstL2ProjectionOperator.cpp
+ * \author Stuart R. Slattery
+ * \brief L2ProjectionOperator unit tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+#include <cstdlib>
+#include <sstream>
+#include <algorithm>
+#include <cassert>
+
+#include "reference_implementation/DTK_ReferenceHexMesh.hpp"
+
+#include <DTK_FieldMultiVector.hpp>
+
+#include <DTK_L2ProjectionOperator.hpp>
+
+#include <DTK_BasicEntityPredicates.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_Array.hpp>
+#include <Teuchos_ParameterList.hpp>
+
+//---------------------------------------------------------------------------//
+// TEST EPSILON
+//---------------------------------------------------------------------------//
+
+const double epsilon = 1.0e-7;
+
+//---------------------------------------------------------------------------//
+// Tests
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( L2ProjectionOperator, l2_projection )
+{
+    using namespace DataTransferKit;
+
+    // Get the communicator.
+    Teuchos::RCP<const Teuchos::Comm<int> > comm =
+	Teuchos::DefaultComm<int>::getComm();
+    DataTransferKit::LocalEntityPredicate local_pred( comm->getRank() );
+
+    // Set the global problem bounds.
+    double x_max = 3.1;
+    double y_max = 5.2;
+    double z_max = 8.3;
+    
+    // Create a source mesh and field.
+    int num_sx = 8;
+    int num_sy = 8;
+    int num_sz = 8;
+    double sx_width = x_max / num_sx;
+    double sy_width = y_max / num_sy;
+    double sz_width = z_max / num_sz;    
+    Teuchos::Array<double> sx( num_sx );
+    Teuchos::Array<double> sy( num_sy );
+    Teuchos::Array<double> sz( num_sz );
+    for ( int i = 0; i < num_sx; ++i ) sx[i] = i*sx_width;
+    for ( int i = 0; i < num_sy; ++i ) sy[i] = i*sy_width;
+    for ( int i = 0; i < num_sz; ++i ) sz[i] = i*sz_width;
```

Comment:
How about a 2nd constructor to `ReferenceHexMesh` that takes the number of separations in each directions or at least a helper function that builds the arrays for you?